classes/core/PKPString.php

```diff
@@ -399,23 +402,68 @@ public static function getAmbiguousExtensionsMap()
      * Strip unsafe HTML from the input text. Covers XSS attacks like scripts,
      * onclick(...) attributes, javascript: urls, and special characters.
      *
-     * @param string $input input string
-     * @param string $configKey The config section key['allowed_html', 'allowed_title_html']
+     * @param string|null   $input input string
+     * @param string        $key   The config section key['allowed_html', 'allowed_title_html']
      *
      * @return string
      */
-    public static function stripUnsafeHtml($input, $configKey = 'allowed_html')
+    public static function stripUnsafeHtml(?string $input, string $key = 'allowed_html'): string
     {
-        static $purifier;
-        if (!isset($purifier)) {
-            $config = HTMLPurifier_Config::createDefault();
-            $config->set('Core.Encoding', 'utf-8');
-            $config->set('HTML.Doctype', 'HTML 4.01 Transitional');
-            $config->set('HTML.Allowed', Config::getVar('security', $configKey));
-            $config->set('Cache.SerializerPath', 'cache');
-            $purifier = new HTMLPurifier($config);
+        if (!$input) {
+            return '';
         }
-        return $purifier->purify((string) $input);
+
+        static $sanitizer;
+        static $configKey;
+        static $allowedTagToAttributeMap;
+
+        if ($configKey !== $key) {
+            $configKey = $key;
+            $allowedTagToAttributeMap = Str::of(Config::getVar('security', $configKey))
+                ->explode(',')
+                ->mapWithKeys(function(string $allowedTagWithAttr) {
+                    
+                    // Extract the tag itself (e.g. div, p, a ...)
+                    preg_match('/\[[^][]+]\K|\w+/', $allowedTagWithAttr, $matches);
+                    $allowedTag = collect($matches)->first();
+
+                    // Extract the attributes associated with tag (e.g. class, href ...)
+                    preg_match("/\[([^\]]*)\]/", $allowedTagWithAttr, $matches);
+                    $allowedAttributes = collect($matches)->last();
+
+                    if($allowedTag) {
+                        return [
+                            $allowedTag => Str::of($allowedAttributes)
+                                ->explode('|')
+                                ->filter()
+                                ->toArray()
+                        ];
+                    }
+            
+                    return [];
+                });
+        }
+
+        if(!isset($sanitizer)) {
```

Comment:
This will instantiate a sanitizer for the first value of `$configKey`, and after that it will be re-used regardless of whether the `$configKey` matches.

Reply:
Moved both the attributes tag map and sanitizer as static cache as both needed at the end .